### PR TITLE
Update generate proof and verify proof - Closes #6587

### LINKED ISSUE #6587: Refactor generateProof & verifyProof to follow LIP appendix
### Description

Refactor standard merkle tree `generateProof` to return and `verifyProof` to accept
```
interface Proof {
	readonly siblingHashes: Buffer[];
	readonly indexes: number[];
	readonly size: number;
}
```

### Acceptance Criteria

- `generateProof` should follow the [LIP-0031](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0031.md) and return above information.
- Test should be updated to expect the above format using the same fixture

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #6587

### How was it solved?

- Update generate proof / verify proof to have interface as defined in LIP
- Use newly created utility functions

### How was it tested?

- Add tests using fixture given by research team

### CODE REVIEW

File: elements/lisk-tree/src/merkle_tree/merkle_tree.ts [hunk 1]
```diff
@@ -512,25 +403,74 @@ export class MerkleTree {
...
+		}
+		return idxs;
+	}
+
+	private async _getAppendPathHashes(): Promise<Buffer[]> {
+		const appendPathNodes = await this._getAppendPathNodes();
+		return appendPathNodes.map(p => p.hash);
+	}
+
+	private async _getNode(nodeHash: Buffer): Promise<NodeInfo> {
```
Comment (shuse2) [inline elements/lisk-tree/src/merkle_tree/merkle_tree.ts:437]: this is just changing to `private` function, and by changing, the location within the file needed to be changed. However, the content is exactly the same

File: elements/lisk-tree/src/merkle_tree/merkle_tree.ts [hunk 2]
```diff
@@ -512,25 +403,74 @@ export class MerkleTree {
...
+				const node = await this._getNode(query);
+				idxs.push(toIndex(node.nodeIndex, node.layerIndex, height));
+			} catch (error) {
+				idxs.push(0);
+			}
+		}
+		return idxs;
+	}
+
+	private async _getAppendPathHashes(): Promise<Buffer[]> {
```
Comment (shuse2) [inline elements/lisk-tree/src/merkle_tree/merkle_tree.ts:432]: this is just changing to private function, and by changing, the location within the file needed to be changed. However, the content is exactly the same

File: elements/lisk-tree/src/merkle_tree/utils.ts [hunk 3]
```diff
@@ -210,7 +144,7 @@ export const treeSortFn = (a: number, b: number) => {
 
 export const insertNewIndex = (arr: number[], val: number) => {
 	const insertIndex = binarySearch(arr, n => treeSortFn(val, n) < 0);
-	if (arr[insertIndex] !== val) {
+	if (arr[insertIndex - 1] !== val) {
```
Comment (ricott1) [inline elements/lisk-tree/src/merkle_tree/utils.ts:147]: mh, is there a reason why this is one off?

**Review by ricott1 — APPROVED** [review]
A small comment that would be nice to clarify about the binarySearch function

File: elements/lisk-tree/src/merkle_tree/utils.ts (hunk 3 above)
Comment (shuse2) [reply elements/lisk-tree/src/merkle_tree/utils.ts:147]: I think it's returning the length? for example,
`[8, 12, 26]`, `12` returns `2`.

**Review by mehmetegemen — APPROVED** [review]
LGTM

**Pull request merged**